src/eth/execute.yaml

```diff
@@ -54,3 +54,24 @@
         gasUsed:
           title: Gas used
           $ref: '#/components/schemas/uint'
+- name: eth_multicall
+  summary: Executes a sequence of message calls building on each other's state without creating transactions on the block chain, optionally overriding block and state data
+  params:
+    - name: Version
```

Comment:
I believe this is because versioning wasn't considered until after-the-fact, and you can't easily bolt-on versioning as a parameter later.  By including versioning in the function from the start, we can do a better job than having it be an afterthought.

That being said, it does appear that the engine API is leaning into this mistake and is now creating new methods with a `V1` suffix.  I would *prefer* that the Ethereum JSON-RPC API (which has a different consumer base from the engine API) went down the road of enveloped requests or discriminable requests rather than versioned method names, but this isn't a deal breaker as long as there is a mechanism for versioning.

Reply:
I think it would be better to come up with uniform versioning scheme for methods, I'm not sure we've looked into this enough yet to commit to this parameter-based versioning.

It would be nice if someone could open an issue to discuss this further. There are several possibilities here:
1) add version parameter to the end - I think this is mostly backward compatible, if it isn't added just assume v1
2) add version parameter at beginning - not backwards compatible, benefit is ???
3) all methods accept 1 object with version as top-level value then args - ???
4) add new method for backwards incompatible changes - similar to engine api
5) new method which says which version the api is and move the entire api forward with new version updates
6) ??? probably a lot more possibilities that haven't been thought through

Reply:
Created: https://github.com/ethereum/execution-apis/issues/408#issue-1716017699